worker/main_test.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"cloud.google.com/go/pubsub"
+	"cloud.google.com/go/pubsub/pstest"
+	"context"
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"google.golang.org/api/option"
+	"google.golang.org/grpc"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+	"time"
+)
+
+func TestGetEnv(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	val := "test"
+	os.Setenv("TEST", val)
+	defaultVal := "default"
+	assert.Equal(val, getEnv("TEST", defaultVal))
+	assert.NotEqual(defaultVal, getEnv("TEST", defaultVal))
+}
+
+func TestGetEnvDefault(t *testing.T) {
+	t.Parallel()
+	assert := assert.New(t)
+
+	defaultVal := "default"
+	assert.Equal(defaultVal, getEnv("test", defaultVal))
+}
+
+func TestSubscribe(t *testing.T) {
+	assert := assert.New(t)
+	//create a fake Pub Sub serer
+	ctx := context.Background()
+	// Start a fake server running locally.
+	srv := pstest.NewServer()
```

Comment:
This server setup code probably wants pulling out into a method if we do any further tests on the pub-sub